azurerm/internal/services/network/virtual_hub_connection_resource.go

```diff
@@ -89,32 +92,18 @@ func resourceArmVirtualHubConnectionCreate(d *schema.ResourceData, meta interfac
 	locks.ByName(id.Name, virtualHubResourceName)
 	defer locks.UnlockByName(id.Name, virtualHubResourceName)
 
-	virtualHub, err := client.Get(ctx, id.ResourceGroup, id.Name)
-	if err != nil {
-		if utils.ResponseWasNotFound(virtualHub.Response) {
-			return fmt.Errorf("Virtual Hub %q was not found in Resource Group %q", id.Name, id.ResourceGroup)
-		}
-
-		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): %+v", id.Name, id.ResourceGroup, err)
-	}
-
-	if virtualHub.VirtualHubProperties == nil {
-		return fmt.Errorf("Error retrieving Virtual Hub %q (Resource Group %q): `properties` was nil", id.Name, id.ResourceGroup)
-	}
-
 	name := d.Get("name").(string)
 
-	if features.ShouldResourcesBeImported() {
-		if connection, _ := findVirtualHubConnection(name, virtualHub); connection != nil {
-			return tf.ImportAsExistsError("azurerm_virtual_hub_connection", *connection.ID)
+	if d.IsNewResource() {
```

Comment:
this method isn't updating - so we can remove this `if` here

Reply:
looks like this is still pending?

Reply:
Now that those two deprecated attributes are not `ForceNew()`, I need to extend the create callback to CreateorUpdate, hence keep this one.